### DISCUSSION TITLE
Get fresh data in production when query doesn't change with useAsyncData (cache not working)

### DISCUSSION THREAD

**Discussion by inthegreenwild** [nuxt/nuxt#29371] [Questions]
I'm working on a Nuxt app that uses useAsyncData to fetch content from a CMS. UseAsyncData hits and endpoints `/api/content/` which then queries the CMS. Occasionally, the content within the CMS will update but that doesn't change the parameters of the request so I'm not sure how to tell Nuxt to request new data after a period of time. I tried watching the video by Alexander Lichter [here](https://www.youtube.com/watch?v=aQPR0xn-MMk) about setting a TTL in the payload, but this doesn't work as expected in production. 

```
getCachedData(key: string, nuxtApp: NuxtApp) {
    const data = nuxtApp.payload.data[key];

    if (!data || !data?.DOCSITE_TTL) {
        return;
    }

    const expirationDate = new Date(data.DOCSITE_TTL);

    // refresh after 60 seconds
    const expiryTime = 60 * 1000;
    expirationDate.setTime(expirationDate.getTime() + expiryTime);
    const isExpired = expirationDate.getTime() < Date.now();

    if (isExpired) {
        clearNuxtData(key);
        return;
    }

    return data;
}
```

```
const {data, error} = await useAsyncData(route.path, () => $fetch('/api/content/content-key'), { getCachedData })
```

On the server I am appending a DOCSITE_TTL date just as described in the video (except I'm doing it on nuxt server endpoint instead of in transform). The problem is that after the initial 60 seconds, getCacheData will ALWAYS return as expired because the payload.data[key] remains the same. It never gets updated on the server, even after adding clearNuxtData. It only updates `window.__NUXT__. payload.` 

Behavior expected: 
1 If a user refreshes the page, get a fresh copy of the data from the nuxt endpoint which queries the CMS
2. If a user navigates around client-side, get a fresh copy of the data from the nuxt endpoint which queries the CMS after 60 seconds

**Comment by xjccc** [comment]
Use top `await` ?

```
const {data, error} = await useAsyncData(route.path, () => $fetch('/api/content/content-key'), { getCachedData })
```

**Reply by inthegreenwild** [reply]
This was actually a typo on my end in the discussion. In the code, we do use await. This doesn't affect the payload data refreshing unfortunately.

**Reply by xjccc** [reply]
Are you update `data.DOCSITE_TTL`

**Reply by inthegreenwild** [reply]
Yes, and the data updates on the client, but not the server payload. So for instance: I can go to a new page on the client, go back with the browser button, and I will get the correct data. On window.__nuxt__ the data will be correct as well. But if I do a page refresh, the old data comes back from the server.
